website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,551 @@
+                }
+            });
+
+            var shrunkSortButtons = ['Asc', 'Desc'];
+            var direction = self.preSelectDirection();
+            document.getElementById(direction).className = 'selected';
+            if (direction === 'Asc'){
+                document.getElementById('Desc').className = 'not-selected'
+            }
+            else {
+                document.getElementById('Asc').className = 'not-selected'
+            }
+        };
+
+        // Shifts all nodes back to master node list, sorts, and returns self.countDisplayed() number of nodes for display
+        self.sortNodesAndModifyDisplay = function () {
+            self.restoreToNodeList(self.displayedNodes());
+            self.sortBySortState();
+            self.colorSortButtons();
+            self.displayedNodes(self.nodes().splice(0, self.countDisplayed()))
+        };
+
+        // Filtering on title
+        self.noTitleMatch = function (node) {
+            return (node.attributes.title.toUpperCase().indexOf(self.filter().toUpperCase()) === -1);
+        };
+
+        // Filtering on contrib
+        self.noContributorMatch = function (node) {
+            var contributors = self.contributorMapping[node.id];
+
+            for (var c = 0; c < contributors.length; c++) {
+                if (contributors[c].toUpperCase().indexOf(self.filter().toUpperCase()) !== -1){
+                    return false
+                }
+            }
+            return true
+        };
+
+        // Filtering on tag
+        self.noTagMatch = function (node) {
+            var tags = node.attributes.tags;
+            for (var t = 0; t < tags.length; t++){
+                if (tags[t].toUpperCase().indexOf(self.filter().toUpperCase()) !== -1) {
+                    return false
+                }
+            }
+            return true
+        };
+
+        // Filters nodes
+        self.filterNodes = function (){
+            for (var n = self.nodes().length - 1; n >= 0; n--) {
+                var node = self.nodes()[n];
+                if (self.noTitleMatch(node) && self.noContributorMatch(node) && self.noTagMatch(node)) {
+                    self.nonMatchingNodes().push(node);
+                    self.nodes().splice(n, 1)
+                }
+            }
+        };
+
+        // Colors node div and changes cursor to pointer on hover
+        self.mouseOver = function (node) {
+            node.style.backgroundColor='#E0EBF3';
+            node.style.cursor = 'pointer'
+        };
+
+        self.mouseOut = function (node) {
+            node.style.backgroundColor='#fcfcfc'
+        };
+
+        self.clearSearch = function () {
+            document.getElementById('searchQuery').value="";
+            self.filter(document.getElementById('searchQuery').value);
+            self.quickSearch()
+        };
+
+        self.quickSearch = function () {
+            self.filter(document.getElementById('searchQuery').value);
```

Comment:
For instance here, the filter value should be an m.prop() so that quick search keyup event or onchange event trigger the m.prop change and the filtering function. That's more of a mithril way without using DOM access either through js or jQuery.